### PR TITLE
DataProtection: Move away from soon to be obsolete APIs

### PR DESCRIPTION
**Pull request by HaoK**
Preparation for runtime obsoleting these APIs in an incoming runtime PR

### CODE REVIEW

**Review by jeffhandley — APPROVED** [review]
LGTM

File: src/DataProtection/DataProtection/src/AuthenticatedEncryption/ManagedAuthenticatedEncryptorFactory.cs [hunk 1]
```diff
@@ -96,14 +96,7 @@ private Func<SymmetricAlgorithm> GetSymmetricBlockCipherAlgorithmFactory(Managed
...
             {
-                Func<Aes>? factory = null;
-                if (OSVersionUtil.IsWindows())
-                {
-                    // If we're on desktop CLR and running on Windows, use the FIPS-compliant implementation.
-                    factory = () => new AesCryptoServiceProvider();
-                }
-
-                return factory ?? Aes.Create;
+                return Aes.Create;
```
Comment (pranavkm) [inline src/DataProtection/DataProtection/src/AuthenticatedEncryption/ManagedAuthenticatedEncryptorFactory.cs:99]: Is the FIPS compliance statement no longer valid?

**Review by GrabYourPitchforks — APPROVED** [review]
LGTM! Keep in mind that since you're cross-compiling for downlevel, if you need to change how hash algorithm factories work, you'll want to ifdef core vs. full framework so that you call the correct overload to maintain FIPS compliance.

For encryption algorithms like AES / 3DES, your PR doesn't affect FIPS compliance, so things look good.

File: src/DataProtection/DataProtection/src/AuthenticatedEncryption/ManagedAuthenticatedEncryptorFactory.cs (hunk 1 above)
Comment (HaoK) [reply src/DataProtection/DataProtection/src/AuthenticatedEncryption/ManagedAuthenticatedEncryptorFactory.cs:99]: AesCryptoServiceProvider is being obsoleted, not sure if the base API is FIPS compliant?

Comment (pranavkm) [reply src/DataProtection/DataProtection/src/AuthenticatedEncryption/ManagedAuthenticatedEncryptorFactory.cs:99]: Yup, but that's in netcore, no? Aes in net461 is probably not FIPs compliant. Do we need to cross-compile this?

**Pull request merged**